src/AaveCurator/AaveV3CuratorPayload.sol

```diff
@@ -0,0 +1,110 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.19;
+
+import {IERC20} from 'solidity-utils/contracts/oz-common/interfaces/IERC20.sol';
+import {IProposalGenericExecutor} from 'aave-helpers/interfaces/IProposalGenericExecutor.sol';
+import {AaveV2Ethereum, AaveV2EthereumAssets} from 'aave-address-book/AaveV2Ethereum.sol';
+import {AaveV2EthereumArc} from 'aave-address-book/AaveV2EthereumArc.sol';
+
+import {AaveCurator} from './AaveCurator.sol';
+import {TokenAddresses} from './TokenAddresses.sol';
+
+/**
+ * @title Launch AaveCurator
+ * @author Llama
+ * Governance: https://governance.aave.com/t/arfc-deploy-ethereum-collector-contract/12205
+ * Snapshot: https://snapshot.org/#/aave.eth/proposal/0xb4141f12f7ec8e037e6320912b5673fcc5909457d9f6201c018d5c15e5aa5083
+ */
+contract AaveV3CuratorPayload is IProposalGenericExecutor {
+    address public constant ARC_aUSDC = 0xd35f648C3C7f17cd1Ba92e5eac991E3EfcD4566d;
+
+  function execute() external {
+    // TODO: How to withdraw from ARC?
+    // AaveV2Ethereum.COLLECTOR.transfer(
+    //   ARC_aUSDC,
+    //   address(this),
+    //   IERC20(ARC_aUSDC).balanceOf(address(AaveV2Ethereum.COLLECTOR))
+    // );
+    // AaveV2EthereumArc.POOL.withdraw(ARC_aUSDC, type(uint256).max, address(this));
+
+    AaveV2Ethereum.COLLECTOR.transfer(
+      AaveV2EthereumAssets.USDC_UNDERLYING,
+      address(this),
+      IERC20(AaveV2EthereumAssets.USDC_UNDERLYING).balanceOf(address(AaveV2Ethereum.COLLECTOR))
+    );
+
+    uint256 balance = IERC20(AaveV2EthereumAssets.USDC_UNDERLYING).balanceOf(address(this));
+
+    IERC20(AaveV2EthereumAssets.USDC_UNDERLYING).approve(address(AaveV2Ethereum.POOL), balance);
+    AaveV2Ethereum.POOL.deposit(
+      AaveV2EthereumAssets.USDC_UNDERLYING,
+      balance,
+      address(AaveV2Ethereum.COLLECTOR),
+      0
+    );
+
+    AaveCurator curator = new AaveCurator();
```

Comment:
I'm quite confused with this.

If you use Initializable (or VersionizedInitializable) for upgradability, you need to use a proxy - how else would you upgrade it?

Reply:
oops, yeah, i meant to do: https://github.com/bgd-labs/aave-proposals/pull/208/files#diff-ad2978446845e4cb62b4e9c7369c7065b0d5cba8427a3bf8502149c0cfd08943R13